manager/controlapi/task.go

```diff
@@ -104,10 +106,18 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 	if request.Filters != nil {
 		tasks = filterTasks(tasks,
 			func(e *api.Task) bool {
-				return filterContains(e.ServiceAnnotations.Name, request.Filters.Names)
+				name := e.Annotations.Name
```

Comment:
This same section of code is present in several places. Let's try to concentrate it somewhere or at least make a note here about why it exists. Unsuspecting developers may look at this an not understand why it exists or when it can be removed.

At some point, this code won't be needed at all.